### PR TITLE
Remove 4000 random tests from System.Globalization.Tests

### PR DESCRIPTION
**Pull request by hughbe**
- Also removes some now unused methods/variables in Compare tests

Before:
```
=== TEST EXECUTION SUMMARY ===
     System.Globalization.Tests  Total: 5247, Errors: 0, Failed: 0, Skipped: 0, Time: 1.159s
```
After:
```
=== TEST EXECUTION SUMMARY ===
     System.Globalization.Tests  Total: 991, Errors: 0, Failed: 0, Skipped: 0, Time: 1.093s
```

Similar to #7218
/cc @stephentoub @tarekgh @weshaggard

### CODE REVIEW

**Comment by stephentoub** [comment]
Are there any interesting test cases we lose because of this that we don't have other tests for?  Does code coverage go down at all?

**Comment by hughbe** [comment]
@stephentoub I don't think coverage is a good indicator here, as CompareInfo delegates the Compare implementation to native platform API calls, which has no coverage.

Also, I can't run a coverage run for System.Globalization, as the code is over at mscorlib

**Comment by stephentoub** [comment]
> I don't think coverage is a good indicator here,

It's one indicator, certainly not the only one. Have you reviewed what kind of data it's being created, and compared that against the other explicit tests that exist?

> I can't run a coverage run for System.Globalization, as the code is over at mscorlib

You can, actually, it's just not as automated and requires building mscorlib locally. You've done that before, haven't you?  If not, you've previously reported coverage changes for types in mscorlib for your previous changes; how did you get that data then? You don't have to for this change, but it would be good to understand if we're missing anything based on manual review.

**Comment by hughbe** [comment]
> You've done that before, haven't you?
Yeah I have - was a *little* reluctant to do that as for some reason I have to uninstall dnx and clear the cache each time I run it - weird little bug

I'll run a coverage run tho

**Comment by hughbe** [comment]
Update: there is no change in coverage
new report
old report

**Comment by hughbe** [comment]
I'll also show you the weird bug I get: 
bug

Running the same test project before copying my custom built mscorlib does not cause any failures

**Comment by tarekgh** [comment]
maybe there is new changes in your custom mscorlib causing this failure. could you send what exactly is failing (I mean the stack of failure in the regex test run)? or you can do ```msbuild /t:rebuildandrun``` from the failed test folder and look at what exactly is failing.

**Comment by stephentoub** [comment]
Most likely the built mscorlib just isn't compatible with the coreclr being used, e.g. you're using a debug mscorlib with a release coreclr.

File: src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs [hunk 1]
```diff
@@ -63,30 +61,6 @@ public static IEnumerable<object[]> IndexOf_TestData()
             yield return new object[] { s_invariantCompare, "cbabababdbaba", "ab", 0, 13, CompareOptions.None, 2 };
         }
 
-        public static IEnumerable<object[]> IndexOf_Random_TestData()
-        {
-            string[] interestingStrings = new string[] { "", "a", "1", "-", "A", "!", "abc", "aBc", "a\u0400Bc", "I", "i", "\u0130", "\u0131", "A", "\uFF21", "\uFE57" };
```
Comment (stephentoub) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs]: Do these strings show up in any of the explicitly-written inputs being used to test IndexOf?  If so, LGTM.

Comment (hughbe) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs]: They don't actually, would you like me to add them?

**Comment by hughbe** [comment]
> Most likely the built mscorlib just isn't compatible with the coreclr being used, e.g. you're using a debug mscorlib with a release coreclr.

That's a likely reason actually, I'll report back if this doesn't work

File: src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs]: Yes, please do.  Some of those values have special meaning, e.g. I suspect "I", "i", 0x130, and 0x131 were all chosen due to relevance to Turkish casing.  One entry for each of the special characters if they don't already exist should be sufficient.

Comment (hughbe) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs]: done in a PR feedback comm

File: src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs [hunk 2]
```diff
@@ -61,30 +59,15 @@ public static IEnumerable<object[]> IndexOf_TestData()
...
 
-        public static IEnumerable<object[]> IndexOf_Random_TestData()
-        {
             string[] interestingStrings = new string[] { "", "a", "1", "-", "A", "!", "abc", "aBc", "a\u0400Bc", "I", "i", "\u0130", "\u0131", "A", "\uFF21", "\uFE57" };
             foreach (string string1 in interestingStrings)
             {
                 foreach (string string2 in interestingStrings)
                 {
-                    yield return new object[] { s_currentCompare, string1, string2, 0, string1.Length, CompareOptions.Ordinal, PredictIndexOfOrdinalResult(string1, string2) };
+                    yield return new object[] { s_currentCompare, string1, string2, 0, string1.Length, CompareOptions.Ordinal, string1.IndexOf(string2, StringComparison.Ordinal) };
```
Comment (stephentoub) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs]: I realize this is what the original test was doing, but this seems busted to me.  We're testing indexOf by telling it the expected position is the result of IndexOf.

What I'd intended in my comments was to just add some entries manually based on these inputs, like the rest of the test data.

Comment (hughbe) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.IndexOf.cs]: Understood. I've fixed this

**Comment by stephentoub** [comment]
LGTM.  Thanks, @hughbe.

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Debug Build and Test please (Jenkins hang)

**Comment by weshaggard** [comment]
Thanks @hughbe for cleaning these up.

**Pull request merged**